src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/LocationAttribute.kt

```diff
@@ -157,32 +157,30 @@ class LocationAttribute private constructor(
         else -> AvailabilityStatus.NONE
       }
 
-      LocationUsage.SCHEDULE -> getScheduleAvailability(probationTeam, startingOnDateTime)
+      LocationUsage.SCHEDULE -> getScheduleAvailability(probationTeam, onDate, startTime, endTime)
       else -> return AvailabilityStatus.NONE
     }
   }
 
   /**
    * We need to look at the schedules as a whole to determine availability. A schedule on its own is not enough.
    */
-  private fun getScheduleAvailability(probationTeam: ProbationTeam, dateAndTime: LocalDateTime): AvailabilityStatus {
-    val schedules = locationSchedule.filter { it.fallsOn(dateAndTime) }
-
+  private fun getScheduleAvailability(probationTeam: ProbationTeam, onDate: LocalDate, startTime: LocalTime, endTime: LocalTime): AvailabilityStatus = run {
     return when {
-      // If there aren't any schedules which fall on the requested date and time then default to SHARED
-      schedules.isEmpty() -> AvailabilityStatus.SHARED
-      // If there are schedules fall on the date and time look and see if any match the following
-      schedules.any { schedule -> schedule.isUsage(LocationScheduleUsage.BLOCKED) } -> AvailabilityStatus.NONE
-      schedules.any { schedule -> schedule.isForProbationTeam(probationTeam) } -> AvailabilityStatus.PROBATION_ROOM
-      schedules.any { schedule -> schedule.isForAnyProbationTeam() } -> AvailabilityStatus.PROBATION_ANY
-      // If there are no matches above then it is not available
-      else -> AvailabilityStatus.NONE
+      // The order in which the checks are carried out is important and must be maintained.
+      locationSchedule.any { it.isSatisfiedBy(OverlappingSpecification(LocationScheduleUsage.BLOCKED, onDate.dayOfWeek, startTime, endTime)) } -> AvailabilityStatus.NONE
+      locationSchedule.any { it.isSatisfiedBy(ProbationTeamSpecification(probationTeam, onDate.dayOfWeek, startTime, endTime)) } -> AvailabilityStatus.PROBATION_ROOM
+      locationSchedule.any { it.isSatisfiedBy(ProbationAnySpecification(onDate.dayOfWeek, startTime, endTime)) } -> AvailabilityStatus.PROBATION_ANY
+      locationSchedule.any { it.isSatisfiedBy(OverlappingSpecification(LocationScheduleUsage.COURT, onDate.dayOfWeek, startTime, endTime)) } -> AvailabilityStatus.NONE
+      locationSchedule.any { it.isSatisfiedBy(OverlappingRoomSpecification(LocationScheduleUsage.PROBATION, onDate.dayOfWeek, startTime, endTime)) } -> AvailabilityStatus.NONE
```

Comment:
One for later perhaps - a comment on the last two items, and their purpose?

Reply:
Yarp